basic_hashtable/b_hashtables.py

```diff
@@ -16,28 +16,45 @@ def __init__(self, key, value):
 class BasicHashTable:
     def __init__(self, capacity):
         self.capacity = capacity
-        self.count = 0
+        self.storage = [None] * capacity
         pass
 
 
 # '''
 # Fill this in.
 # Research and implement the djb2 hash function
 # '''
-def hash(string, max):
-    hashed = 5381
-    for i in string:
-        hashed = ((hashed << 0) + hashed) + ord(string) 
-    return hash & 0xFFFFFFFF
-
+def hash(string):
+    hash = 5381
 
+    for i in string:
+        hash = (((hash << 5) + hash) + ord(i)) & 0xFFFFFFFF
+    return hash
 # '''
 # Fill this in.
 
 # If you are overwriting a value with a different key, print a warning.
 # '''
+
+
 def hash_table_insert(hash_table, key, value):
-    pass
+    key_hash = hash(key)
+    key_value = Pair(key, value)
+
+    # to get the index you need take the has modulus by the size(capcity)
+    index = key_hash % hash_table.capacity
+    # we want to check if the storage is not None
+    if hash_table.storage[index] is not None:
+        # If the key matches the key in the storage
+        if key == hash_table.storage[index].key:
```

Comment:
This should be the following because if the key is the same your code should not print a collision error message.

```
if key != hash_table.storage[index].key:
```

Reply:
This line was to check if it comes in and that Key and it's None then check to see if this key already exists.  So it shouldn't be key !=hash_table.storage[index].key. I believe it should be to see if we already have the key then we can print the collision we talked about it in the class.

Reply:
A collision does not occur for similar keys only when you are trying to overwrite a pair with another pair with a unique key.